contracts/tokenomics/generator/tests/integration.rs

```diff
@@ -50,6 +57,272 @@ struct PoolWithProxy {
+        &mut app,
+        &helper_controller.factory,
+        None,
+        None,
+        [
+            AssetInfo::Token {
+                contract_addr: cny_token.clone(),
+            },
+            AssetInfo::Token {
+                contract_addr: eur_token.clone(),
+            },
+        ],
+    );
+
+    register_lp_tokens_in_generator(
+        &mut app,
+        &helper_controller.generator,
+        vec![PoolWithProxy {
+            pool: (lp_cny_eur.to_string(), Uint128::from(100u32)),
+            proxy: None,
+        }],
+    );
+
+    // Mint tokens, so user can deposit
+    mint_tokens(&mut app, pair_cny_eur.clone(), &lp_cny_eur, &user1, 10);
+
+    // Create short lock user1
+    helper_controller
+        .escrow_helper
+        .mint_xastro(&mut app, USER1, 100);
+
+    helper_controller
+        .escrow_helper
+        .create_lock(&mut app, USER1, WEEK * 3, 100f32)
+        .unwrap();
+
+    deposit_lp_tokens_to_generator(
+        &mut app,
+        &helper_controller.generator,
+        USER1,
+        &[(&lp_cny_eur, 10)],
+    );
+
+    check_token_balance(&mut app, &lp_cny_eur, &helper_controller.generator, 10);
+
+    // check if virtual amount equal to 10
+    check_emission_balance(
+        &mut app,
+        &helper_controller.generator,
+        &lp_cny_eur,
+        &user1,
+        10,
+    );
+
+    // Mint tokens, so user2 can deposit
+    mint_tokens(&mut app, pair_cny_eur.clone(), &lp_cny_eur, &user2, 10);
+
+    // Create short lock user2
+    helper_controller
+        .escrow_helper
+        .mint_xastro(&mut app, USER2, 100);
+
+    helper_controller
+        .escrow_helper
+        .create_lock(&mut app, USER2, WEEK * 3, 100f32)
+        .unwrap();
+
+    deposit_lp_tokens_to_generator(
+        &mut app,
+        &helper_controller.generator,
+        USER2,
+        &[(&lp_cny_eur, 10)],
+    );
+
+    check_token_balance(&mut app, &lp_cny_eur, &helper_controller.generator, 20);
+
+    // check if virtual amount equal to 10
+    check_emission_balance(
+        &mut app,
+        &helper_controller.generator,
+        &lp_cny_eur,
+        &user2,
+        10,
+    );
+
+    let err = app
+        .execute_contract(
+            Addr::unchecked(USER1),
+            helper_controller.generator.clone(),
+            &ExecuteMsg::CheckpointUserBoost {
+                generators: vec![lp_cny_eur.to_string(); 26],
+            },
+            &[],
+        )
+        .unwrap_err();
+    assert_eq!("Maximum generator limit exceeded!", err.to_string());
+
+    // check user1's ASTRO balance
+    check_token_balance(
+        &mut app,
+        &helper_controller.escrow_helper.astro_token,
+        &user1,
+        0,
+    );
+
+    // check user2's ASTRO balance
+    check_token_balance(
+        &mut app,
+        &helper_controller.escrow_helper.astro_token,
+        &user2,
+        0,
+    );
+
+    app.next_block(WEEK);
+    app.update_block(|bi| bi.time = bi.time.plus_seconds(WEEK));
```

Comment:
app.next_block is just helper over app.update_block. If you want to shift by 2 weeks here simply use app.next_block(2*WEEK). If you do not need this then just remove this line (app.update_block(...)...).